{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create files mapping FACS well and plate to basic allele metadata\n",
    "\n",
    "This script takes cherry picking and allele info files as input and produces a table relating FACS well and plate IDs to allele IDs and symbols. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import polars as pl\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dualipa_inputs = \"../1_inputs\"\n",
    "dualipa_outputs = \"../3_outputs\"\n",
    "allele_collection_inputs = \"../../../1_allele_collection/1_inputs\"\n",
    "allele_collection_outputs = \"../../../1_allele_collection/3_outputs\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Vidal lab ran experiment once and analyzed the data. Some alleles didn't pass the qa/qc thresholds or got no data. Batch5 is is a re-run for all alleles that failed in the first set of batches. Here, we process everything together from scratch. An allele should always be compared to the wt on the same batch and plate. The summary stats can then be compared across batches. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (2_311, 6)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>orf_id</th><th>mut_id</th><th>node_type</th><th>dest_pla_id</th><th>dest_pla</th><th>dest_well</th></tr><tr><td>i64</td><td>i64</td><td>str</td><td>i64</td><td>str</td><td>str</td></tr></thead><tbody><tr><td>56360</td><td>0</td><td>&quot;wt&quot;</td><td>23</td><td>&quot;VUSDUALXP_23&quot;</td><td>&quot;A12&quot;</td></tr><tr><td>56360</td><td>0</td><td>&quot;wt&quot;</td><td>23</td><td>&quot;VUSDUALXP_23&quot;</td><td>&quot;B12&quot;</td></tr><tr><td>56360</td><td>0</td><td>&quot;wt&quot;</td><td>23</td><td>&quot;VUSDUALXP_23&quot;</td><td>&quot;C12&quot;</td></tr><tr><td>56360</td><td>0</td><td>&quot;wt&quot;</td><td>23</td><td>&quot;VUSDUALXP_23&quot;</td><td>&quot;D12&quot;</td></tr><tr><td>9905</td><td>0</td><td>&quot;wt&quot;</td><td>23</td><td>&quot;VUSDUALXP_23&quot;</td><td>&quot;F02&quot;</td></tr><tr><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td></tr><tr><td>null</td><td>null</td><td>&quot;pdest_empty&quot;</td><td>20</td><td>&quot;VUSDUALXP_20&quot;</td><td>&quot;A08&quot;</td></tr><tr><td>null</td><td>null</td><td>&quot;pdest_empty&quot;</td><td>21</td><td>&quot;VUSDUALXP_21&quot;</td><td>&quot;A07&quot;</td></tr><tr><td>null</td><td>null</td><td>&quot;pdest_empty&quot;</td><td>21</td><td>&quot;VUSDUALXP_21&quot;</td><td>&quot;A08&quot;</td></tr><tr><td>null</td><td>null</td><td>&quot;pdest_empty&quot;</td><td>22</td><td>&quot;VUSDUALXP_22&quot;</td><td>&quot;A07&quot;</td></tr><tr><td>null</td><td>null</td><td>&quot;pdest_empty&quot;</td><td>22</td><td>&quot;VUSDUALXP_22&quot;</td><td>&quot;A08&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (2_311, 6)\n",
       "┌────────┬────────┬─────────────┬─────────────┬──────────────┬───────────┐\n",
       "│ orf_id ┆ mut_id ┆ node_type   ┆ dest_pla_id ┆ dest_pla     ┆ dest_well │\n",
       "│ ---    ┆ ---    ┆ ---         ┆ ---         ┆ ---          ┆ ---       │\n",
       "│ i64    ┆ i64    ┆ str         ┆ i64         ┆ str          ┆ str       │\n",
       "╞════════╪════════╪═════════════╪═════════════╪══════════════╪═══════════╡\n",
       "│ 56360  ┆ 0      ┆ wt          ┆ 23          ┆ VUSDUALXP_23 ┆ A12       │\n",
       "│ 56360  ┆ 0      ┆ wt          ┆ 23          ┆ VUSDUALXP_23 ┆ B12       │\n",
       "│ 56360  ┆ 0      ┆ wt          ┆ 23          ┆ VUSDUALXP_23 ┆ C12       │\n",
       "│ 56360  ┆ 0      ┆ wt          ┆ 23          ┆ VUSDUALXP_23 ┆ D12       │\n",
       "│ 9905   ┆ 0      ┆ wt          ┆ 23          ┆ VUSDUALXP_23 ┆ F02       │\n",
       "│ …      ┆ …      ┆ …           ┆ …           ┆ …            ┆ …         │\n",
       "│ null   ┆ null   ┆ pdest_empty ┆ 20          ┆ VUSDUALXP_20 ┆ A08       │\n",
       "│ null   ┆ null   ┆ pdest_empty ┆ 21          ┆ VUSDUALXP_21 ┆ A07       │\n",
       "│ null   ┆ null   ┆ pdest_empty ┆ 21          ┆ VUSDUALXP_21 ┆ A08       │\n",
       "│ null   ┆ null   ┆ pdest_empty ┆ 22          ┆ VUSDUALXP_22 ┆ A07       │\n",
       "│ null   ┆ null   ┆ pdest_empty ┆ 22          ┆ VUSDUALXP_22 ┆ A08       │\n",
       "└────────┴────────┴─────────────┴─────────────┴──────────────┴───────────┘"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in cherry picking files for both the initial batches and the second round\n",
    "cp_col = ['orf_id', 'mut_id', 'node_type','dest_pla_id', 'dest_pla', 'dest_well']\n",
    "\n",
    "pdest_layout_initial_batches_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP.csv', index_col=0)[cp_col]\n",
    "pdest_layout_b5_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP_batch5.csv', index_col=0)[cp_col]\n",
    "\n",
    "pdest_layout_df = pd.concat([pdest_layout_b5_df, pdest_layout_initial_batches_df])\n",
    "pdest_layout_df = pl.DataFrame(pdest_layout_df)\n",
    "pdest_layout_df = pdest_layout_df.with_columns(\n",
    "    pl.col(\"orf_id\").cast(pl.Int64).alias(\"orf_id\"),\n",
    "    pl.col(\"mut_id\").cast(pl.Int64).alias(\"mut_id\")\n",
    ")\n",
    "pdest_layout_df\n",
    "### This doesn't have well A01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>perfectly validated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>partially validated, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>wild type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>partial wild type, target not covered, &gt;= 50% ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>target and off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>truncated, &lt; 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>99</td>\n",
       "      <td>no reads</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id                                        description\n",
       "0   1                                perfectly validated\n",
       "1   2               partially validated, >= 50% coverage\n",
       "2   3                                          wild type\n",
       "3   4  partial wild type, target not covered, >= 50% ...\n",
       "4   5    target and off-target mutation, >= 50% coverage\n",
       "5   6               off-target mutation, >= 50% coverage\n",
       "6   7                          truncated, < 50% coverage\n",
       "7  99                                           no reads"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seq_confirm_code_df = pd.read_table(f\"{allele_collection_inputs}/sequence_confirmation_class_code.tsv\")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_code_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>orf_id</th>\n",
       "      <th>mut_id</th>\n",
       "      <th>ccsb_mutation_id</th>\n",
       "      <th>ccsb_allele_id</th>\n",
       "      <th>spdi</th>\n",
       "      <th>nt_change</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>collection</th>\n",
       "      <th>entry_plate_orig</th>\n",
       "      <th>entry_well_orig</th>\n",
       "      <th>entry_plate_conso</th>\n",
       "      <th>entry_well_conso</th>\n",
       "      <th>entry_seq_pool</th>\n",
       "      <th>db_plate</th>\n",
       "      <th>db_well</th>\n",
       "      <th>n2h_plate</th>\n",
       "      <th>n2h_well</th>\n",
       "      <th>dualip_plate</th>\n",
       "      <th>dualip_well</th>\n",
       "      <th>mislocalization_plate</th>\n",
       "      <th>mislocalization_well</th>\n",
       "      <th>entry_sequenced</th>\n",
       "      <th>entry_sequence_confirmation_class</th>\n",
       "      <th>db_sequenced</th>\n",
       "      <th>db_sequence_confirmation_class</th>\n",
       "      <th>n2h_sequenced</th>\n",
       "      <th>n2h_sequence_confirmation_class</th>\n",
       "      <th>n2h_ref_sequence_confirmation_class</th>\n",
       "      <th>dualip_sequenced</th>\n",
       "      <th>dualip_sequence_confirmation_class</th>\n",
       "      <th>dualip_ref_sequence_confirmation_class</th>\n",
       "      <th>mislocalization_sequenced</th>\n",
       "      <th>mislocalization_sequence_confirmation_class</th>\n",
       "      <th>mislocalization_ref_sequence_confirmation_class</th>\n",
       "      <th>gene_variant</th>\n",
       "      <th>chr_num</th>\n",
       "      <th>nuc_loc</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>Chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>CCSBVarC000001</td>\n",
       "      <td>ALE0000584</td>\n",
       "      <td>NC_000001.11:155240033:C:G</td>\n",
       "      <td>160G&gt;C</td>\n",
       "      <td>Val54Leu</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>F12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBA1_Val54Leu</td>\n",
       "      <td>1</td>\n",
       "      <td>155240033</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>73</td>\n",
       "      <td>CCSBVarC000002</td>\n",
       "      <td>ALE00000002</td>\n",
       "      <td>NC_000001.11:155238225:G:A</td>\n",
       "      <td>670C&gt;T</td>\n",
       "      <td>Leu224Phe</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>C12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBA1_Leu224Phe</td>\n",
       "      <td>1</td>\n",
       "      <td>155238225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>113</td>\n",
       "      <td>CCSBVarC000003</td>\n",
       "      <td>ALE00000003</td>\n",
       "      <td>NC_000001.11:155237453:C:T</td>\n",
       "      <td>887G&gt;A</td>\n",
       "      <td>Arg296Gln</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>D12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBA1_Arg296Gln</td>\n",
       "      <td>1</td>\n",
       "      <td>155237453</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>231</td>\n",
       "      <td>CCSBVarC000004</td>\n",
       "      <td>ALE00000004</td>\n",
       "      <td>NC_000001.11:155235252:A:G</td>\n",
       "      <td>1448T&gt;C</td>\n",
       "      <td>Leu483Pro</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>E12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBA1_Leu483Pro</td>\n",
       "      <td>1</td>\n",
       "      <td>155235252</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>213510</td>\n",
       "      <td>CCSBVarC003869</td>\n",
       "      <td>ALE00003869</td>\n",
       "      <td>NC_000001.11:155239934:G:A</td>\n",
       "      <td>259C&gt;T</td>\n",
       "      <td>Arg87Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>B02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBA1_Arg87Trp</td>\n",
       "      <td>1</td>\n",
       "      <td>155239934</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7493</th>\n",
       "      <td>SRY</td>\n",
       "      <td>ENSG00000184895</td>\n",
       "      <td>100070152</td>\n",
       "      <td>229149</td>\n",
       "      <td>CCSBVarC007069</td>\n",
       "      <td>ALE00007067</td>\n",
       "      <td>NC_000024.10:2787412:C:T</td>\n",
       "      <td>192G&gt;A</td>\n",
       "      <td>Met64Ile</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>D05</td>\n",
       "      <td>GDEhDisVCh_40029</td>\n",
       "      <td>D09</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRY_Met64Ile</td>\n",
       "      <td>24</td>\n",
       "      <td>2787412</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7494</th>\n",
       "      <td>SRY</td>\n",
       "      <td>ENSG00000184895</td>\n",
       "      <td>100070152</td>\n",
       "      <td>229151</td>\n",
       "      <td>CCSBVarC007072</td>\n",
       "      <td>ALE00007070</td>\n",
       "      <td>NC_000024.10:2787267:C:T</td>\n",
       "      <td>337G&gt;A</td>\n",
       "      <td>Ala113Thr</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>F09</td>\n",
       "      <td>GDEhDisVCh_40029</td>\n",
       "      <td>H09</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRY_Ala113Thr</td>\n",
       "      <td>24</td>\n",
       "      <td>2787267</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7495</th>\n",
       "      <td>SRY</td>\n",
       "      <td>ENSG00000184895</td>\n",
       "      <td>100070152</td>\n",
       "      <td>229152</td>\n",
       "      <td>CCSBVarC007068</td>\n",
       "      <td>ALE00007066</td>\n",
       "      <td>NC_000024.10:2787207:G:A</td>\n",
       "      <td>397C&gt;T</td>\n",
       "      <td>Arg133Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>E09</td>\n",
       "      <td>GDEhDisVCh_40029</td>\n",
       "      <td>G09</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRY_Arg133Trp</td>\n",
       "      <td>24</td>\n",
       "      <td>2787207</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7496</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>ENSG00000168036</td>\n",
       "      <td>100070227</td>\n",
       "      <td>212487</td>\n",
       "      <td>CCSBVarC007073</td>\n",
       "      <td>ALE00007071</td>\n",
       "      <td>NC_000003.12:41225785:A:G</td>\n",
       "      <td>860A&gt;G</td>\n",
       "      <td>Asn287Ser</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>B07</td>\n",
       "      <td>GDEhDisVCh_40034</td>\n",
       "      <td>E09</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTNNB1_Asn287Ser</td>\n",
       "      <td>3</td>\n",
       "      <td>41225785</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7497</th>\n",
       "      <td>TBX19</td>\n",
       "      <td>ENSG00000143178</td>\n",
       "      <td>100070258</td>\n",
       "      <td>212489</td>\n",
       "      <td>CCSBVarC007075</td>\n",
       "      <td>ALE00007073</td>\n",
       "      <td>NC_000001.11:168291213:T:G</td>\n",
       "      <td>257T&gt;G</td>\n",
       "      <td>Met86Arg</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>GDEhDisVCh_40046</td>\n",
       "      <td>F02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TBX19_Met86Arg</td>\n",
       "      <td>1</td>\n",
       "      <td>168291213</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7498 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      symbol  ensembl_gene_id     orf_id  mut_id ccsb_mutation_id  \\\n",
       "0       GBA1  ENSG00000177628          2       6   CCSBVarC000001   \n",
       "1       GBA1  ENSG00000177628          2      73   CCSBVarC000002   \n",
       "2       GBA1  ENSG00000177628          2     113   CCSBVarC000003   \n",
       "3       GBA1  ENSG00000177628          2     231   CCSBVarC000004   \n",
       "4       GBA1  ENSG00000177628          2  213510   CCSBVarC003869   \n",
       "...      ...              ...        ...     ...              ...   \n",
       "7493     SRY  ENSG00000184895  100070152  229149   CCSBVarC007069   \n",
       "7494     SRY  ENSG00000184895  100070152  229151   CCSBVarC007072   \n",
       "7495     SRY  ENSG00000184895  100070152  229152   CCSBVarC007068   \n",
       "7496  CTNNB1  ENSG00000168036  100070227  212487   CCSBVarC007073   \n",
       "7497   TBX19  ENSG00000143178  100070258  212489   CCSBVarC007075   \n",
       "\n",
       "     ccsb_allele_id                        spdi nt_change  aa_change  \\\n",
       "0        ALE0000584  NC_000001.11:155240033:C:G    160G>C   Val54Leu   \n",
       "1       ALE00000002  NC_000001.11:155238225:G:A    670C>T  Leu224Phe   \n",
       "2       ALE00000003  NC_000001.11:155237453:C:T    887G>A  Arg296Gln   \n",
       "3       ALE00000004  NC_000001.11:155235252:A:G   1448T>C  Leu483Pro   \n",
       "4       ALE00003869  NC_000001.11:155239934:G:A    259C>T   Arg87Trp   \n",
       "...             ...                         ...       ...        ...   \n",
       "7493    ALE00007067    NC_000024.10:2787412:C:T    192G>A   Met64Ile   \n",
       "7494    ALE00007070    NC_000024.10:2787267:C:T    337G>A  Ala113Thr   \n",
       "7495    ALE00007066    NC_000024.10:2787207:G:A    397C>T  Arg133Trp   \n",
       "7496    ALE00007071   NC_000003.12:41225785:A:G    860A>G  Asn287Ser   \n",
       "7497    ALE00007073  NC_000001.11:168291213:T:G    257T>G   Met86Arg   \n",
       "\n",
       "     collection  entry_plate_orig entry_well_orig entry_plate_conso  \\\n",
       "0           RC4  RC4_Mut_GDEh1026             H01  GDEhDisVCh_40054   \n",
       "1           RC4  RC4_Mut_GDEh1026             E01  GDEhDisVCh_40054   \n",
       "2           RC4  RC4_Mut_GDEh1026             F01  GDEhDisVCh_40054   \n",
       "3           RC4  RC4_Mut_GDEh1026             G01  GDEhDisVCh_40054   \n",
       "4         CEGS2   CegsMutGDEh1035             B03  GDEhDisVCh_40054   \n",
       "...         ...               ...             ...               ...   \n",
       "7493      CEGS2   CegsMutGDEh1023             D05  GDEhDisVCh_40029   \n",
       "7494      CEGS2   CegsMutGDEh1023             F09  GDEhDisVCh_40029   \n",
       "7495      CEGS2   CegsMutGDEh1023             E09  GDEhDisVCh_40029   \n",
       "7496      CEGS2   CegsMutGDEh1023             B07  GDEhDisVCh_40034   \n",
       "7497      CEGS2   CegsMutGDEh1043             G08  GDEhDisVCh_40046   \n",
       "\n",
       "     entry_well_conso  entry_seq_pool          db_plate db_well n2h_plate  \\\n",
       "0                 F12             2.0  RC4_Mut_GDDh1026     H01       NaN   \n",
       "1                 C12             2.0  RC4_Mut_GDDh1026     E01       NaN   \n",
       "2                 D12             2.0  RC4_Mut_GDDh1026     F01       NaN   \n",
       "3                 E12             2.0  RC4_Mut_GDDh1026     G01       NaN   \n",
       "4                 B02             2.0   CegsMutGDDh1035     B03       NaN   \n",
       "...               ...             ...               ...     ...       ...   \n",
       "7493              D09             1.0               NaN     NaN       NaN   \n",
       "7494              H09             1.0               NaN     NaN       NaN   \n",
       "7495              G09             1.0               NaN     NaN       NaN   \n",
       "7496              E09             2.0               NaN     NaN       NaN   \n",
       "7497              F02             2.0   CegsMutGDDh1043     G08       NaN   \n",
       "\n",
       "     n2h_well dualip_plate dualip_well mislocalization_plate  \\\n",
       "0         NaN          NaN         NaN                   NaN   \n",
       "1         NaN          NaN         NaN                   NaN   \n",
       "2         NaN          NaN         NaN                   NaN   \n",
       "3         NaN          NaN         NaN                   NaN   \n",
       "4         NaN          NaN         NaN                   NaN   \n",
       "...       ...          ...         ...                   ...   \n",
       "7493      NaN          NaN         NaN                   NaN   \n",
       "7494      NaN          NaN         NaN                   NaN   \n",
       "7495      NaN          NaN         NaN                   NaN   \n",
       "7496      NaN          NaN         NaN                   NaN   \n",
       "7497      NaN          NaN         NaN                   NaN   \n",
       "\n",
       "     mislocalization_well  entry_sequenced  entry_sequence_confirmation_class  \\\n",
       "0                     NaN                1                                1.0   \n",
       "1                     NaN                1                                1.0   \n",
       "2                     NaN                1                                1.0   \n",
       "3                     NaN                1                                1.0   \n",
       "4                     NaN                1                                1.0   \n",
       "...                   ...              ...                                ...   \n",
       "7493                  NaN                1                                1.0   \n",
       "7494                  NaN                1                                1.0   \n",
       "7495                  NaN                1                                1.0   \n",
       "7496                  NaN                1                                1.0   \n",
       "7497                  NaN                1                                1.0   \n",
       "\n",
       "      db_sequenced  db_sequence_confirmation_class  n2h_sequenced  \\\n",
       "0                1                             2.0              0   \n",
       "1                1                             1.0              0   \n",
       "2                1                             7.0              0   \n",
       "3                1                             2.0              0   \n",
       "4                1                             2.0              0   \n",
       "...            ...                             ...            ...   \n",
       "7493             0                             NaN              0   \n",
       "7494             0                             NaN              0   \n",
       "7495             0                             NaN              0   \n",
       "7496             0                             NaN              0   \n",
       "7497             1                             1.0              0   \n",
       "\n",
       "      n2h_sequence_confirmation_class  n2h_ref_sequence_confirmation_class  \\\n",
       "0                                 NaN                                  NaN   \n",
       "1                                 NaN                                  NaN   \n",
       "2                                 NaN                                  NaN   \n",
       "3                                 NaN                                  NaN   \n",
       "4                                 NaN                                  NaN   \n",
       "...                               ...                                  ...   \n",
       "7493                              NaN                                  NaN   \n",
       "7494                              NaN                                  NaN   \n",
       "7495                              NaN                                  NaN   \n",
       "7496                              NaN                                  NaN   \n",
       "7497                              NaN                                  NaN   \n",
       "\n",
       "      dualip_sequenced  dualip_sequence_confirmation_class  \\\n",
       "0                    0                                 NaN   \n",
       "1                    0                                 NaN   \n",
       "2                    0                                 NaN   \n",
       "3                    0                                 NaN   \n",
       "4                    0                                 NaN   \n",
       "...                ...                                 ...   \n",
       "7493                 0                                 NaN   \n",
       "7494                 0                                 NaN   \n",
       "7495                 0                                 NaN   \n",
       "7496                 0                                 NaN   \n",
       "7497                 0                                 NaN   \n",
       "\n",
       "      dualip_ref_sequence_confirmation_class  mislocalization_sequenced  \\\n",
       "0                                        NaN                          0   \n",
       "1                                        NaN                          0   \n",
       "2                                        NaN                          0   \n",
       "3                                        NaN                          0   \n",
       "4                                        NaN                          0   \n",
       "...                                      ...                        ...   \n",
       "7493                                     NaN                          0   \n",
       "7494                                     NaN                          0   \n",
       "7495                                     NaN                          0   \n",
       "7496                                     NaN                          0   \n",
       "7497                                     NaN                          0   \n",
       "\n",
       "      mislocalization_sequence_confirmation_class  \\\n",
       "0                                             NaN   \n",
       "1                                             NaN   \n",
       "2                                             NaN   \n",
       "3                                             NaN   \n",
       "4                                             NaN   \n",
       "...                                           ...   \n",
       "7493                                          NaN   \n",
       "7494                                          NaN   \n",
       "7495                                          NaN   \n",
       "7496                                          NaN   \n",
       "7497                                          NaN   \n",
       "\n",
       "      mislocalization_ref_sequence_confirmation_class      gene_variant  \\\n",
       "0                                                 NaN     GBA1_Val54Leu   \n",
       "1                                                 NaN    GBA1_Leu224Phe   \n",
       "2                                                 NaN    GBA1_Arg296Gln   \n",
       "3                                                 NaN    GBA1_Leu483Pro   \n",
       "4                                                 NaN     GBA1_Arg87Trp   \n",
       "...                                               ...               ...   \n",
       "7493                                              NaN      SRY_Met64Ile   \n",
       "7494                                              NaN     SRY_Ala113Thr   \n",
       "7495                                              NaN     SRY_Arg133Trp   \n",
       "7496                                              NaN  CTNNB1_Asn287Ser   \n",
       "7497                                              NaN    TBX19_Met86Arg   \n",
       "\n",
       "      chr_num    nuc_loc ref_allele alt_allele Chrom  \n",
       "0           1  155240033          C          G     1  \n",
       "1           1  155238225          G          A     1  \n",
       "2           1  155237453          C          T     1  \n",
       "3           1  155235252          A          G     1  \n",
       "4           1  155239934          G          A     1  \n",
       "...       ...        ...        ...        ...   ...  \n",
       "7493       24    2787412          C          T     Y  \n",
       "7494       24    2787267          C          T     Y  \n",
       "7495       24    2787207          G          A     Y  \n",
       "7496        3   41225785          A          G     3  \n",
       "7497        1  168291213          T          G     1  \n",
       "\n",
       "[7498 rows x 42 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (7_498, 42)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>symbol</th><th>ensembl_gene_id</th><th>orf_id</th><th>mut_id</th><th>ccsb_mutation_id</th><th>ccsb_allele_id</th><th>spdi</th><th>nt_change</th><th>aa_change</th><th>collection</th><th>entry_plate_orig</th><th>entry_well_orig</th><th>entry_plate_conso</th><th>entry_well_conso</th><th>entry_seq_pool</th><th>db_plate</th><th>db_well</th><th>n2h_plate</th><th>n2h_well</th><th>dualip_plate</th><th>dualip_well</th><th>mislocalization_plate</th><th>mislocalization_well</th><th>entry_sequenced</th><th>entry_sequence_confirmation_class</th><th>db_sequenced</th><th>db_sequence_confirmation_class</th><th>n2h_sequenced</th><th>n2h_sequence_confirmation_class</th><th>n2h_ref_sequence_confirmation_class</th><th>dualip_sequenced</th><th>dualip_sequence_confirmation_class</th><th>dualip_ref_sequence_confirmation_class</th><th>mislocalization_sequenced</th><th>mislocalization_sequence_confirmation_class</th><th>mislocalization_ref_sequence_confirmation_class</th><th>gene_variant</th><th>chr_num</th><th>nuc_loc</th><th>ref_allele</th><th>alt_allele</th><th>Chrom</th></tr><tr><td>str</td><td>str</td><td>i64</td><td>i64</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>f64</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>i64</td><td>f64</td><td>i64</td><td>f64</td><td>i64</td><td>f64</td><td>f64</td><td>i64</td><td>f64</td><td>f64</td><td>i64</td><td>f64</td><td>f64</td><td>str</td><td>i64</td><td>i64</td><td>str</td><td>str</td><td>str</td></tr></thead><tbody><tr><td>&quot;GBA1&quot;</td><td>&quot;ENSG00000177628&quot;</td><td>2</td><td>6</td><td>&quot;CCSBVarC000001&quot;</td><td>&quot;ALE0000584&quot;</td><td>&quot;NC_000001.11:155240033:C:G&quot;</td><td>&quot;160G&gt;C&quot;</td><td>&quot;Val54Leu&quot;</td><td>&quot;RC4&quot;</td><td>&quot;RC4_Mut_GDEh1026&quot;</td><td>&quot;H01&quot;</td><td>&quot;GDEhDisVCh_40054&quot;</td><td>&quot;F12&quot;</td><td>2.0</td><td>&quot;RC4_Mut_GDDh1026&quot;</td><td>&quot;H01&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>2.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;GBA1_Val54Leu&quot;</td><td>1</td><td>155240033</td><td>&quot;C&quot;</td><td>&quot;G&quot;</td><td>&quot;1&quot;</td></tr><tr><td>&quot;GBA1&quot;</td><td>&quot;ENSG00000177628&quot;</td><td>2</td><td>73</td><td>&quot;CCSBVarC000002&quot;</td><td>&quot;ALE00000002&quot;</td><td>&quot;NC_000001.11:155238225:G:A&quot;</td><td>&quot;670C&gt;T&quot;</td><td>&quot;Leu224Phe&quot;</td><td>&quot;RC4&quot;</td><td>&quot;RC4_Mut_GDEh1026&quot;</td><td>&quot;E01&quot;</td><td>&quot;GDEhDisVCh_40054&quot;</td><td>&quot;C12&quot;</td><td>2.0</td><td>&quot;RC4_Mut_GDDh1026&quot;</td><td>&quot;E01&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;GBA1_Leu224Phe&quot;</td><td>1</td><td>155238225</td><td>&quot;G&quot;</td><td>&quot;A&quot;</td><td>&quot;1&quot;</td></tr><tr><td>&quot;GBA1&quot;</td><td>&quot;ENSG00000177628&quot;</td><td>2</td><td>113</td><td>&quot;CCSBVarC000003&quot;</td><td>&quot;ALE00000003&quot;</td><td>&quot;NC_000001.11:155237453:C:T&quot;</td><td>&quot;887G&gt;A&quot;</td><td>&quot;Arg296Gln&quot;</td><td>&quot;RC4&quot;</td><td>&quot;RC4_Mut_GDEh1026&quot;</td><td>&quot;F01&quot;</td><td>&quot;GDEhDisVCh_40054&quot;</td><td>&quot;D12&quot;</td><td>2.0</td><td>&quot;RC4_Mut_GDDh1026&quot;</td><td>&quot;F01&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>7.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;GBA1_Arg296Gln&quot;</td><td>1</td><td>155237453</td><td>&quot;C&quot;</td><td>&quot;T&quot;</td><td>&quot;1&quot;</td></tr><tr><td>&quot;GBA1&quot;</td><td>&quot;ENSG00000177628&quot;</td><td>2</td><td>231</td><td>&quot;CCSBVarC000004&quot;</td><td>&quot;ALE00000004&quot;</td><td>&quot;NC_000001.11:155235252:A:G&quot;</td><td>&quot;1448T&gt;C&quot;</td><td>&quot;Leu483Pro&quot;</td><td>&quot;RC4&quot;</td><td>&quot;RC4_Mut_GDEh1026&quot;</td><td>&quot;G01&quot;</td><td>&quot;GDEhDisVCh_40054&quot;</td><td>&quot;E12&quot;</td><td>2.0</td><td>&quot;RC4_Mut_GDDh1026&quot;</td><td>&quot;G01&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>2.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;GBA1_Leu483Pro&quot;</td><td>1</td><td>155235252</td><td>&quot;A&quot;</td><td>&quot;G&quot;</td><td>&quot;1&quot;</td></tr><tr><td>&quot;GBA1&quot;</td><td>&quot;ENSG00000177628&quot;</td><td>2</td><td>213510</td><td>&quot;CCSBVarC003869&quot;</td><td>&quot;ALE00003869&quot;</td><td>&quot;NC_000001.11:155239934:G:A&quot;</td><td>&quot;259C&gt;T&quot;</td><td>&quot;Arg87Trp&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1035&quot;</td><td>&quot;B03&quot;</td><td>&quot;GDEhDisVCh_40054&quot;</td><td>&quot;B02&quot;</td><td>2.0</td><td>&quot;CegsMutGDDh1035&quot;</td><td>&quot;B03&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>2.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;GBA1_Arg87Trp&quot;</td><td>1</td><td>155239934</td><td>&quot;G&quot;</td><td>&quot;A&quot;</td><td>&quot;1&quot;</td></tr><tr><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td></tr><tr><td>&quot;SRY&quot;</td><td>&quot;ENSG00000184895&quot;</td><td>100070152</td><td>229149</td><td>&quot;CCSBVarC007069&quot;</td><td>&quot;ALE00007067&quot;</td><td>&quot;NC_000024.10:2787412:C:T&quot;</td><td>&quot;192G&gt;A&quot;</td><td>&quot;Met64Ile&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1023&quot;</td><td>&quot;D05&quot;</td><td>&quot;GDEhDisVCh_40029&quot;</td><td>&quot;D09&quot;</td><td>1.0</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;SRY_Met64Ile&quot;</td><td>24</td><td>2787412</td><td>&quot;C&quot;</td><td>&quot;T&quot;</td><td>&quot;Y&quot;</td></tr><tr><td>&quot;SRY&quot;</td><td>&quot;ENSG00000184895&quot;</td><td>100070152</td><td>229151</td><td>&quot;CCSBVarC007072&quot;</td><td>&quot;ALE00007070&quot;</td><td>&quot;NC_000024.10:2787267:C:T&quot;</td><td>&quot;337G&gt;A&quot;</td><td>&quot;Ala113Thr&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1023&quot;</td><td>&quot;F09&quot;</td><td>&quot;GDEhDisVCh_40029&quot;</td><td>&quot;H09&quot;</td><td>1.0</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;SRY_Ala113Thr&quot;</td><td>24</td><td>2787267</td><td>&quot;C&quot;</td><td>&quot;T&quot;</td><td>&quot;Y&quot;</td></tr><tr><td>&quot;SRY&quot;</td><td>&quot;ENSG00000184895&quot;</td><td>100070152</td><td>229152</td><td>&quot;CCSBVarC007068&quot;</td><td>&quot;ALE00007066&quot;</td><td>&quot;NC_000024.10:2787207:G:A&quot;</td><td>&quot;397C&gt;T&quot;</td><td>&quot;Arg133Trp&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1023&quot;</td><td>&quot;E09&quot;</td><td>&quot;GDEhDisVCh_40029&quot;</td><td>&quot;G09&quot;</td><td>1.0</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;SRY_Arg133Trp&quot;</td><td>24</td><td>2787207</td><td>&quot;G&quot;</td><td>&quot;A&quot;</td><td>&quot;Y&quot;</td></tr><tr><td>&quot;CTNNB1&quot;</td><td>&quot;ENSG00000168036&quot;</td><td>100070227</td><td>212487</td><td>&quot;CCSBVarC007073&quot;</td><td>&quot;ALE00007071&quot;</td><td>&quot;NC_000003.12:41225785:A:G&quot;</td><td>&quot;860A&gt;G&quot;</td><td>&quot;Asn287Ser&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1023&quot;</td><td>&quot;B07&quot;</td><td>&quot;GDEhDisVCh_40034&quot;</td><td>&quot;E09&quot;</td><td>2.0</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;CTNNB1_Asn287Ser&quot;</td><td>3</td><td>41225785</td><td>&quot;A&quot;</td><td>&quot;G&quot;</td><td>&quot;3&quot;</td></tr><tr><td>&quot;TBX19&quot;</td><td>&quot;ENSG00000143178&quot;</td><td>100070258</td><td>212489</td><td>&quot;CCSBVarC007075&quot;</td><td>&quot;ALE00007073&quot;</td><td>&quot;NC_000001.11:168291213:T:G&quot;</td><td>&quot;257T&gt;G&quot;</td><td>&quot;Met86Arg&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;CegsMutGDEh1043&quot;</td><td>&quot;G08&quot;</td><td>&quot;GDEhDisVCh_40046&quot;</td><td>&quot;F02&quot;</td><td>2.0</td><td>&quot;CegsMutGDDh1043&quot;</td><td>&quot;G08&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>null</td><td>1</td><td>1.0</td><td>1</td><td>1.0</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>0</td><td>null</td><td>null</td><td>&quot;TBX19_Met86Arg&quot;</td><td>1</td><td>168291213</td><td>&quot;T&quot;</td><td>&quot;G&quot;</td><td>&quot;1&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (7_498, 42)\n",
       "┌────────┬─────────────────┬───────────┬────────┬───┬───────────┬────────────┬────────────┬───────┐\n",
       "│ symbol ┆ ensembl_gene_id ┆ orf_id    ┆ mut_id ┆ … ┆ nuc_loc   ┆ ref_allele ┆ alt_allele ┆ Chrom │\n",
       "│ ---    ┆ ---             ┆ ---       ┆ ---    ┆   ┆ ---       ┆ ---        ┆ ---        ┆ ---   │\n",
       "│ str    ┆ str             ┆ i64       ┆ i64    ┆   ┆ i64       ┆ str        ┆ str        ┆ str   │\n",
       "╞════════╪═════════════════╪═══════════╪════════╪═══╪═══════════╪════════════╪════════════╪═══════╡\n",
       "│ GBA1   ┆ ENSG00000177628 ┆ 2         ┆ 6      ┆ … ┆ 155240033 ┆ C          ┆ G          ┆ 1     │\n",
       "│ GBA1   ┆ ENSG00000177628 ┆ 2         ┆ 73     ┆ … ┆ 155238225 ┆ G          ┆ A          ┆ 1     │\n",
       "│ GBA1   ┆ ENSG00000177628 ┆ 2         ┆ 113    ┆ … ┆ 155237453 ┆ C          ┆ T          ┆ 1     │\n",
       "│ GBA1   ┆ ENSG00000177628 ┆ 2         ┆ 231    ┆ … ┆ 155235252 ┆ A          ┆ G          ┆ 1     │\n",
       "│ GBA1   ┆ ENSG00000177628 ┆ 2         ┆ 213510 ┆ … ┆ 155239934 ┆ G          ┆ A          ┆ 1     │\n",
       "│ …      ┆ …               ┆ …         ┆ …      ┆ … ┆ …         ┆ …          ┆ …          ┆ …     │\n",
       "│ SRY    ┆ ENSG00000184895 ┆ 100070152 ┆ 229149 ┆ … ┆ 2787412   ┆ C          ┆ T          ┆ Y     │\n",
       "│ SRY    ┆ ENSG00000184895 ┆ 100070152 ┆ 229151 ┆ … ┆ 2787267   ┆ C          ┆ T          ┆ Y     │\n",
       "│ SRY    ┆ ENSG00000184895 ┆ 100070152 ┆ 229152 ┆ … ┆ 2787207   ┆ G          ┆ A          ┆ Y     │\n",
       "│ CTNNB1 ┆ ENSG00000168036 ┆ 100070227 ┆ 212487 ┆ … ┆ 41225785  ┆ A          ┆ G          ┆ 3     │\n",
       "│ TBX19  ┆ ENSG00000143178 ┆ 100070258 ┆ 212489 ┆ … ┆ 168291213 ┆ T          ┆ G          ┆ 1     │\n",
       "└────────┴─────────────────┴───────────┴────────┴───┴───────────┴────────────┴────────────┴───────┘"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seq_confirm_res_df = pd.read_table(f\"{allele_collection_outputs}/VarChampSeqConfirmationResult_Alleles_Pass_QC.tsv\")\n",
    "seq_confirm_res_df = seq_confirm_res_df.rename(\n",
    "    {\n",
    "        \"orf_id_wt\": \"orf_id\",\n",
    "        \"mutation_id_old\": \"mut_id\"\n",
    "    }, \n",
    "    axis=1\n",
    ")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_res_df)\n",
    "\n",
    "seq_confirm_res_df = pl.DataFrame(seq_confirm_res_df)\n",
    "display(seq_confirm_res_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "shape: (9,)\n",
      "Series: 'dualip_sequence_confirmation_class' [f64]\n",
      "[\n",
      "\tnull\n",
      "\t1.0\n",
      "\t2.0\n",
      "\t3.0\n",
      "\t4.0\n",
      "\t5.0\n",
      "\t6.0\n",
      "\t7.0\n",
      "\t99.0\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "pdest_layout_seq_conf_df = pdest_layout_df.join(\n",
    "    seq_confirm_res_df, \n",
    "    on=[\"orf_id\",\"mut_id\"],\n",
    "    how=\"left\"\n",
    ")\n",
    "print(pdest_layout_seq_conf_df[\"dualip_sequence_confirmation_class\"].unique())\n",
    "\n",
    "pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.with_columns(\n",
    "    pl.col(\"aa_change\").fill_null(\"WT\"),\n",
    "    pl.col(\"nt_change\").fill_null(\"WT\")\n",
    ")\n",
    "\n",
    "pdest_layout_seq_conf_df.write_csv(f'{dualipa_outputs}/dualipa_experimental_layout.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get allele metadata\n",
    "# allele_info_df = pd.read_table(f\"{allele_collection_inputs}/VarChamp_Consolidated_allele_collection.tsv\")\n",
    "# other_cols = [i for i in allele_info_df.columns if \"mutation_id_\" not in i]\n",
    "# mut_cols = [i for i in allele_info_df.columns if \"mutation_id_\" in i]\n",
    "# allele_info_df = allele_info_df.melt(id_vars=other_cols, \n",
    "#                                      value_vars=mut_cols, \n",
    "#                                      var_name='mut_type', \n",
    "#                                      value_name='mut_id')\n",
    "# allele_info_df\n",
    "# Get gene-level metadata\n",
    "# gene_info_df = pd.read_csv(f\"{allele_collection_outputs}/orf_info.csv\")\n",
    "# gene_info_df = gene_info_df[['orf_id', 'symbol']]\n",
    "# orf_to_symbol = gene_info_df.set_index(\"orf_id\")[\"symbol\"].to_dict()\n",
    "# orf_to_symbol = {str(k): v for k, v in orf_to_symbol.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(pdest_layout_seq_conf_df[\"orf_id\"].unique())\n",
    "\n",
    "## Pandas version:\n",
    "## QC with allele collection df for consistency\n",
    "## Skipped for future\n",
    "# meta_cols = ['orf_id', 'mut_id', 'aa_change', 'nt_change', 'symbol']\n",
    "# # pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.merge(allele_info_df[meta_cols], on=['orf_id', 'mut_id'], how='left', suffixes=(\"\", \"_ac\"))\n",
    "# # pdest_layout_seq_conf_df\n",
    "# # assert(all(pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol\"]==pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol_ac\"]))\n",
    "# # assert(all(pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change_ac\"]))\n",
    "# # assert(all(pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change_ac\"]))\n",
    "# # pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.drop(columns=[\"symbol_ac\",\"aa_change_ac\",\"nt_change_ac\"])\n",
    "\n",
    "# ## replace NAs\n",
    "# pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].astype(\"Int64\").astype(str)\n",
    "# pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].replace(\"<NA>\", np.nan)\n",
    "# pdest_layout_seq_conf_df\n",
    "\n",
    "# ## Replace missing symbols\n",
    "# pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]] = pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]].fillna(\"WT\")\n",
    "# pdest_layout_seq_conf_df[\"symbol\"] = pdest_layout_seq_conf_df[\"symbol\"].fillna(pdest_layout_seq_conf_df[\"orf_id\"].map(orf_to_symbol))\n",
    "# pdest_layout_seq_conf_df.to_csv(f'{dualipa_outputs}/dualipa_experimental_layout.csv', index=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "varchamp",
   "language": "python",
   "name": "varchamp"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
