{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create edgotype profiles\n",
    "\n",
    "In this step, we make one edgotype profile per gene of interactors by alleles. Later, we will explore metrics that summarize overall perturbation for each allele (quantitative score like imaging / abundance)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import polars as pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "inputs_dir = \"../1_inputs/sqY2H\"\n",
    "outputs_dir = \"../3_outputs/sqY2H\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (4_426, 20)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>ad_orf_id</th><th>db_orf_id</th><th>db_mut_id</th><th>retest_batch</th><th>normalized_score</th><th>assay_id</th><th>seq_ad</th><th>seq_db</th><th>db_sequenced</th><th>db_sequence_confirmation_class</th><th>ad_symbol</th><th>symbol</th><th>aa_change</th><th>nt_change</th><th>ensembl_gene_id</th><th>collection</th><th>clinvar_clnsig_clean</th><th>gnomad_af</th><th>StarStatus</th><th>allele_0</th></tr><tr><td>i64</td><td>i64</td><td>i64</td><td>str</td><td>f64</td><td>i64</td><td>i64</td><td>i64</td><td>f64</td><td>f64</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>f64</td><td>f64</td><td>str</td></tr></thead><tbody><tr><td>14221</td><td>14221</td><td>218283</td><td>&quot;VUSAPWT1B2&quot;</td><td>0.9375</td><td>1</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;KCTD1&quot;</td><td>&quot;KCTD1&quot;</td><td>&quot;Gly62Asp&quot;</td><td>&quot;185G&gt;A&quot;</td><td>&quot;ENSG00000134504&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;1_Pathogenic&quot;</td><td>null</td><td>1.0</td><td>&quot;KCTD1_Gly62Asp&quot;</td></tr><tr><td>2098</td><td>71642</td><td>0</td><td>&quot;VUSAPWT1B2&quot;</td><td>0.4375</td><td>1</td><td>1</td><td>1</td><td>null</td><td>null</td><td>&quot;EFHC2&quot;</td><td>&quot;CACNB4&quot;</td><td>null</td><td>null</td><td>&quot;ENSG00000182389&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>&quot;CACNB4&quot;</td></tr><tr><td>8357</td><td>71001</td><td>205675</td><td>&quot;VUSAPWT6B1&quot;</td><td>0.75</td><td>6</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;ERP27&quot;</td><td>&quot;UBQLN2&quot;</td><td>&quot;Pro506Thr&quot;</td><td>&quot;1516C&gt;A&quot;</td><td>&quot;ENSG00000188021&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;1_Pathogenic&quot;</td><td>null</td><td>1.0</td><td>&quot;UBQLN2_Pro506Thr&quot;</td></tr><tr><td>13850</td><td>71001</td><td>205675</td><td>&quot;VUSAPWT6B1&quot;</td><td>0.5625</td><td>6</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;CSN2&quot;</td><td>&quot;UBQLN2&quot;</td><td>&quot;Pro506Thr&quot;</td><td>&quot;1516C&gt;A&quot;</td><td>&quot;ENSG00000188021&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;1_Pathogenic&quot;</td><td>null</td><td>1.0</td><td>&quot;UBQLN2_Pro506Thr&quot;</td></tr><tr><td>100066551</td><td>7652</td><td>1806</td><td>&quot;VUSAPWT6B1&quot;</td><td>0.0</td><td>6</td><td>1</td><td>1</td><td>1.0</td><td>2.0</td><td>&quot;MAPK1&quot;</td><td>&quot;STXBP1&quot;</td><td>&quot;Asp238Glu&quot;</td><td>&quot;714C&gt;A&quot;</td><td>&quot;ENSG00000136854&quot;</td><td>&quot;Edgotyping3&quot;</td><td>&quot;4_VUS&quot;</td><td>null</td><td>1.0</td><td>&quot;STXBP1_Asp238Glu&quot;</td></tr><tr><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td></tr><tr><td>53964</td><td>2713</td><td>200850</td><td>&quot;VUSAPWT1B1&quot;</td><td>0.625</td><td>1</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;NFIX&quot;</td><td>&quot;MLH1&quot;</td><td>&quot;Ser295Thr&quot;</td><td>&quot;884G&gt;C&quot;</td><td>&quot;ENSG00000076242&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;4_VUS&quot;</td><td>null</td><td>3.0</td><td>&quot;MLH1_Ser295Thr&quot;</td></tr><tr><td>100066360</td><td>11443</td><td>0</td><td>&quot;VUSAPWT1B2&quot;</td><td>0.8125</td><td>1</td><td>1</td><td>1</td><td>null</td><td>null</td><td>&quot;KRT27&quot;</td><td>&quot;CCBE1&quot;</td><td>null</td><td>null</td><td>&quot;ENSG00000183287&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>&quot;CCBE1&quot;</td></tr><tr><td>71001</td><td>52920</td><td>2662</td><td>&quot;VUSAPWT1B1&quot;</td><td>1.0</td><td>1</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;UBQLN2&quot;</td><td>&quot;LITAF&quot;</td><td>&quot;Arg160His&quot;</td><td>&quot;479G&gt;A&quot;</td><td>&quot;ENSG00000189067&quot;</td><td>&quot;Edgotyping3&quot;</td><td>&quot;4_VUS&quot;</td><td>null</td><td>2.0</td><td>&quot;LITAF_Arg160His&quot;</td></tr><tr><td>2265</td><td>10122</td><td>0</td><td>&quot;VUSAPWT1B2&quot;</td><td>0.75</td><td>1</td><td>1</td><td>1</td><td>null</td><td>null</td><td>&quot;CCDC36&quot;</td><td>&quot;EXOC8&quot;</td><td>null</td><td>null</td><td>&quot;ENSG00000116903&quot;</td><td>null</td><td>null</td><td>null</td><td>null</td><td>&quot;EXOC8&quot;</td></tr><tr><td>2778</td><td>71191</td><td>205704</td><td>&quot;VUSAPWT2B1&quot;</td><td>0.4375</td><td>2</td><td>1</td><td>1</td><td>1.0</td><td>1.0</td><td>&quot;SNF8&quot;</td><td>&quot;SUOX&quot;</td><td>&quot;Ala265Asp&quot;</td><td>&quot;794C&gt;A&quot;</td><td>&quot;ENSG00000139531&quot;</td><td>&quot;CEGS2&quot;</td><td>&quot;4_VUS&quot;</td><td>0.000042</td><td>1.0</td><td>&quot;SUOX_Ala265Asp&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (4_426, 20)\n",
       "┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐\n",
       "│ ad_orf_id ┆ db_orf_id ┆ db_mut_id ┆ retest_ba ┆ … ┆ clinvar_c ┆ gnomad_af ┆ StarStatu ┆ allele_0 │\n",
       "│ ---       ┆ ---       ┆ ---       ┆ tch       ┆   ┆ lnsig_cle ┆ ---       ┆ s         ┆ ---      │\n",
       "│ i64       ┆ i64       ┆ i64       ┆ ---       ┆   ┆ an        ┆ f64       ┆ ---       ┆ str      │\n",
       "│           ┆           ┆           ┆ str       ┆   ┆ ---       ┆           ┆ f64       ┆          │\n",
       "│           ┆           ┆           ┆           ┆   ┆ str       ┆           ┆           ┆          │\n",
       "╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡\n",
       "│ 14221     ┆ 14221     ┆ 218283    ┆ VUSAPWT1B ┆ … ┆ 1_Pathoge ┆ null      ┆ 1.0       ┆ KCTD1_Gl │\n",
       "│           ┆           ┆           ┆ 2         ┆   ┆ nic       ┆           ┆           ┆ y62Asp   │\n",
       "│ 2098      ┆ 71642     ┆ 0         ┆ VUSAPWT1B ┆ … ┆ null      ┆ null      ┆ null      ┆ CACNB4   │\n",
       "│           ┆           ┆           ┆ 2         ┆   ┆           ┆           ┆           ┆          │\n",
       "│ 8357      ┆ 71001     ┆ 205675    ┆ VUSAPWT6B ┆ … ┆ 1_Pathoge ┆ null      ┆ 1.0       ┆ UBQLN2_P │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆ nic       ┆           ┆           ┆ ro506Thr │\n",
       "│ 13850     ┆ 71001     ┆ 205675    ┆ VUSAPWT6B ┆ … ┆ 1_Pathoge ┆ null      ┆ 1.0       ┆ UBQLN2_P │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆ nic       ┆           ┆           ┆ ro506Thr │\n",
       "│ 100066551 ┆ 7652      ┆ 1806      ┆ VUSAPWT6B ┆ … ┆ 4_VUS     ┆ null      ┆ 1.0       ┆ STXBP1_A │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆           ┆           ┆           ┆ sp238Glu │\n",
       "│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │\n",
       "│ 53964     ┆ 2713      ┆ 200850    ┆ VUSAPWT1B ┆ … ┆ 4_VUS     ┆ null      ┆ 3.0       ┆ MLH1_Ser │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆           ┆           ┆           ┆ 295Thr   │\n",
       "│ 100066360 ┆ 11443     ┆ 0         ┆ VUSAPWT1B ┆ … ┆ null      ┆ null      ┆ null      ┆ CCBE1    │\n",
       "│           ┆           ┆           ┆ 2         ┆   ┆           ┆           ┆           ┆          │\n",
       "│ 71001     ┆ 52920     ┆ 2662      ┆ VUSAPWT1B ┆ … ┆ 4_VUS     ┆ null      ┆ 2.0       ┆ LITAF_Ar │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆           ┆           ┆           ┆ g160His  │\n",
       "│ 2265      ┆ 10122     ┆ 0         ┆ VUSAPWT1B ┆ … ┆ null      ┆ null      ┆ null      ┆ EXOC8    │\n",
       "│           ┆           ┆           ┆ 2         ┆   ┆           ┆           ┆           ┆          │\n",
       "│ 2778      ┆ 71191     ┆ 205704    ┆ VUSAPWT2B ┆ … ┆ 4_VUS     ┆ 0.000042  ┆ 1.0       ┆ SUOX_Ala │\n",
       "│           ┆           ┆           ┆ 1         ┆   ┆           ┆           ┆           ┆ 265Asp   │\n",
       "└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scores = pl.read_csv(f\"{outputs_dir}/4_normalized_consensus_scores.csv\")\n",
    "scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "scores = pl.read_csv(f\"{outputs_dir}/4_normalized_consensus_scores.csv\")\n",
    "\n",
    "# keep only scores where there is at least one matched reference-variant set\n",
    "orfs_with_alleles = scores.filter(pl.col(\"db_mut_id\") != 0).select(\"db_orf_id\").unique().to_series().to_list()\n",
    "orfs = scores.filter(pl.col(\"db_mut_id\") == 0).select(\"db_orf_id\").unique().to_series().to_list()\n",
    "\n",
    "complete_orf_pairs = [i for i in orfs if i in orfs_with_alleles]\n",
    "scores = scores.filter(pl.col(\"db_orf_id\").is_in(complete_orf_pairs))\n",
    "genes = scores.select(\"db_orf_id\").to_series().unique().to_list()\n",
    "\n",
    "# print(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create profiles for each profiled gene\n",
    "\n",
    "profiles = {}\n",
    "shape_summary = []\n",
    "for gene in genes:\n",
    "    gene_scores = scores.filter(pl.col(\"db_orf_id\") == gene)\n",
    "    profile = gene_scores.pivot(\n",
    "                values=\"normalized_score\",\n",
    "                index=\"ad_orf_id\",\n",
    "                on=\"db_mut_id\"\n",
    "            )\n",
    "    profiles[gene] = profile\n",
    "\n",
    "    shape_summary.append({\n",
    "        \"gene\": gene,\n",
    "        \"n_interactors\": profile.height,\n",
    "        \"n_alleles\": profile.width\n",
    "    })\n",
    "\n",
    "    profile.write_csv(f\"{outputs_dir}/edgotype_profiles/orf_id_{gene}.csv\")\n",
    "\n",
    "shape_summary = pl.DataFrame(shape_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Add a perturbation category for each edge to use for the N2H validation\n",
    "\n",
    "# Extract WT scores\n",
    "wt_scores = scores.filter(pl.col(\"db_mut_id\") == 0).select([\"db_orf_id\", \"ad_orf_id\", \"normalized_score\"])\n",
    "\n",
    "# Merge WT scores back onto the main df\n",
    "scores_with_wt = scores.join(wt_scores, on=[\"db_orf_id\", \"ad_orf_id\"], how=\"left\", suffix=\"_wt\")\n",
    "\n",
    "# Compute allele/WT ratio\n",
    "scores_with_wt = scores_with_wt.with_columns(\n",
    "    (pl.col(\"normalized_score\") / pl.col(\"normalized_score_wt\")).alias(\"ratio\")\n",
    ")\n",
    "\n",
    "# Determine if the allele is perturbed - binary output\n",
    "threshold = 0.5\n",
    "scores_with_wt = scores_with_wt.with_columns(\n",
    "    ((pl.col(\"db_mut_id\") != 0) & (pl.col(\"ratio\") < threshold)).alias(\"perturbed\")\n",
    ")\n",
    "\n",
    "# Build perturbation_classes expression\n",
    "bins = [\n",
    "    (1.0, float(\"inf\"), \"increased edge\"),\n",
    "    (0.8, 1.0, \"similar\"),\n",
    "    (0.6, 0.8, \"slightly perturbed\"),\n",
    "    (0.4, 0.6, \"perturbed\"),\n",
    "    (0.2, 0.4, \"highly perturbed\"),\n",
    "    (0.0, 0.2, \"very highly perturbed\"),\n",
    "]\n",
    "\n",
    "perturbation_expr = pl.when(pl.col(\"db_mut_id\") == 0).then(pl.lit(\"WT\"))\n",
    "\n",
    "perturbation_expr = perturbation_expr.when(pl.col(\"normalized_score_wt\") == 0.0).then(pl.lit(\"WT edge lost\"))\n",
    "\n",
    "perturbation_expr = perturbation_expr.when(pl.col(\"ratio\") == 0.0).then(pl.lit(\"edge lost\"))\n",
    "\n",
    "perturbation_expr = perturbation_expr.when(pl.col(\"ratio\") == 1.0).then(pl.lit(\"WT-like\"))\n",
    "\n",
    "for lower, upper, label in bins:\n",
    "    perturbation_expr = perturbation_expr.when((pl.col(\"ratio\") > lower) & (pl.col(\"ratio\") <= upper)).then(pl.lit(label))\n",
    "\n",
    "\n",
    "perturbation_expr = perturbation_expr.otherwise(pl.lit(\"unclassified\"))\n",
    "\n",
    "scores_with_wt = scores_with_wt.with_columns(\n",
    "    perturbation_expr.alias(\"perturbation_classes\")\n",
    ")\n",
    "\n",
    "scores_with_perturbed = scores_with_wt.select(\"*\")\n",
    "\n",
    "cols_to_keep = ['ad_orf_id',\n",
    " 'db_orf_id',\n",
    " 'db_mut_id',\n",
    " 'normalized_score',\n",
    " 'ad_symbol',\n",
    " 'symbol',\n",
    " 'clinvar_clnsig_clean',\n",
    " 'normalized_score_wt',\n",
    " 'ratio',\n",
    " 'perturbed',\n",
    " 'perturbation_classes']\n",
    "\n",
    "scores_with_perturbed = scores_with_perturbed.select(cols_to_keep)\n",
    "\n",
    "scores_with_perturbed.write_csv(f\"{outputs_dir}/7_edge_perturbation_categorical.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Examine profile characteristics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26\n",
      "shape: (1, 1)\n",
      "┌───────────┐\n",
      "│ n_alleles │\n",
      "│ ---       │\n",
      "│ i64       │\n",
      "╞═══════════╡\n",
      "│ 258       │\n",
      "└───────────┘\n"
     ]
    }
   ],
   "source": [
    "# Number of profiled alleles with more than 5 interactors\n",
    "hq_orfs = shape_summary.filter(pl.col(\"n_interactors\") >= 5)\n",
    "num_orfs = hq_orfs.shape[0]\n",
    "print(num_orfs)\n",
    "\n",
    "print(hq_orfs.select(pl.col(\"n_alleles\")).sum() - num_orfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "84\n",
      "shape: (1, 1)\n",
      "┌───────────┐\n",
      "│ n_alleles │\n",
      "│ ---       │\n",
      "│ i64       │\n",
      "╞═══════════╡\n",
      "│ 659       │\n",
      "└───────────┘\n"
     ]
    }
   ],
   "source": [
    "# Total number of profiled alleles\n",
    "prof_genes = shape_summary.shape[0]\n",
    "print(prof_genes)\n",
    "\n",
    "print(shape_summary.select(pl.col(\"n_alleles\")).sum() - prof_genes)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
