{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count assayed genes and alleles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import polars as pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "inputs_dir = \"../1_inputs/sqY2H\"\n",
    "outputs_dir = \"../3_outputs/sqY2H\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 3466 interactions between 368 interactor proteins and 558 variants from 82 genes.\n",
      "2402 interactions of the 3466 tested (69%) were greater than 0.\n",
      "There are 434 unique PPIs assayed\n"
     ]
    }
   ],
   "source": [
    "# only look in the LWH1 media (r03)\n",
    "lwh1_scores = pl.read_csv(f\"{outputs_dir}/3_normalized_qaqc_consensus_scores.csv\").filter(pl.col(\"condition_name\") == \"LWH1\")\n",
    "\n",
    "# keep only scores where there is at least one matched reference-variant set\n",
    "orfs_with_alleles = lwh1_scores.filter(pl.col(\"db_mut_id\") != 0).select(\"db_orf_id\").unique().to_series().to_list()\n",
    "orfs = lwh1_scores.filter(pl.col(\"db_mut_id\") == 0).select(\"db_orf_id\").unique().to_series().to_list()\n",
    "complete_orf_pairs = [i for i in orfs if i in orfs_with_alleles]\n",
    "lwh1_scores = lwh1_scores.filter(pl.col(\"db_orf_id\").is_in(complete_orf_pairs))\n",
    "\n",
    "# count interactions\n",
    "num_interactions = lwh1_scores.shape[0]\n",
    "num_interactors = len(lwh1_scores.select(\"ad_orf_id\").to_series().unique().to_list())\n",
    "num_genes = len(lwh1_scores.select(\"db_orf_id\").to_series().unique().to_list())\n",
    "num_alleles = len(lwh1_scores.select(\"db_mut_id\").to_series().unique().to_list())\n",
    "\n",
    "num_pos = lwh1_scores.filter(pl.col(\"adjusted_score\") > 0).shape[0]\n",
    "hit_percent = round((num_pos/num_interactions)*100)\n",
    "\n",
    "print(f\"There are {num_interactions} interactions between {num_interactors} interactor proteins and {num_alleles} variants from {num_genes} genes.\")\n",
    "print(f\"{num_pos} interactions of the {num_interactions} tested ({hit_percent}%) were greater than 0.\")\n",
    "\n",
    "# Look at unique gene-gene interactions\n",
    "lwh1_wt = lwh1_scores.filter(pl.col(\"db_mut_id\") == 0).with_columns(\n",
    "        pl.concat_str([pl.col(\"ad_orf_id\"), pl.col(\"db_orf_id\")], separator=\"_\").alias(\"unique_ppi\")\n",
    "    ).select(\"unique_ppi\").to_series().unique()\n",
    "\n",
    "num_unique_ppi = lwh1_wt.shape[0]\n",
    "print(f\"There are {num_unique_ppi} unique PPIs assayed\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using sequence confirmed results:\n",
    "\n",
    "- There are 3466 interactions between 368 interactor proteins and 558 variants from 82 genes.\n",
    "- 2402 interactions of the 3466 tested (69%) were greater than 0.\n",
    "- There are 434 unique PPIs assayed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using raw scores pulled from the database (v1):\n",
    "- There are 2899 interactions between 246 interactor proteins and 678 variants from 81 genes.\n",
    "- 1913 interactions of the 2899 tested (66%) were greater than 0.\n",
    "- There are 319 unique PPIs assayed"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
